client/src/schema/query.ts

```diff
@@ -10,43 +11,121 @@ export type QueryOptions<T> = {
+
+/**
+ * @maxProperties 1
+ * @minProperties 1
+ */
+export type FilterPredicateOp = {
+  $any?: FilterPredicate[];
+  $all?: FilterPredicate[];
+  $none?: FilterPredicate | FilterPredicate[];
+  $not?: FilterPredicate | FilterPredicate[];
+  $is?: FilterValue | FilterValue[];
+  $isNot?: FilterValue | FilterValue[];
+  $lt?: FilterRangeValue;
+  $le?: FilterRangeValue;
+  $gt?: FilterRangeValue;
+  $ge?: FilterRangeValue;
+  $contains?: string;
+  $startsWith?: string;
+  $endsWith?: string;
+  $pattern?: string;
+};
+
+export type FilterPredicateRangeOp = {
+  $lt?: FilterRangeValue;
+  $le?: FilterRangeValue;
+  $gt?: FilterRangeValue;
+  $ge?: FilterRangeValue;
+};
 
-export class Query<T extends XataRecord, R extends XataRecord = T> implements BasePage<T, R> {
-  table: string;
-  repository: Repository<T>;
+export type FilterRangeValue = number | string;
+
+export type FilterValue = number | string | boolean;
+
+export type SortExpression =
+  | string[]
+  | {
+      [key: string]: SortOrder;
+    }
+  | {
+      [key: string]: SortOrder;
+    }[];
+
+export type SortOrder = 'asc' | 'desc';
+
+export type PageConfig = {
+  after?: string;
+  before?: string;
+  first?: string;
+  last?: string;
+  size?: number;
+  offset?: number;
+};
+
+export type ColumnsFilter = string[];
+
+// TODO: Remove all these types with API Schemas PR
+export type QueryTableData = {
+  filter: FilterExpression;
+  sort?: SortExpression;
+  page?: PageConfig;
+  columns?: ColumnsFilter;
+};
 
-  readonly $any?: QueryOrConstraint<T, R>[];
-  readonly $all?: QueryOrConstraint<T, R>[];
-  readonly $not?: QueryOrConstraint<T, R>[];
-  readonly $none?: QueryOrConstraint<T, R>[];
-  readonly $sort?: Record<string, SortDirection>;
-  readonly columns: SelectableColumn<T>[] = ['*'];
+export class Query<T extends XataRecord, R extends XataRecord = T> {
+  #table: string;
+  #repository: Repository<T>;
+  #data: QueryTableData = { filter: {} };
 
-  // Cursor pagination
-  readonly query: Query<T, R> = this;
-  readonly meta: PaginationQueryMeta = { page: { cursor: 'start', more: true } };
-  readonly records: R[] = [];
+  constructor(
+    repository: Repository<T> | null,
+    table: string,
+    data: Partial<QueryTableData>,
+    parent?: Partial<QueryTableData>
+  ) {
+    this.#table = table;
 
-  constructor(repository: Repository<T> | null, table: string, data: Partial<Query<T, R>>, parent?: Query<T, R>) {
     if (repository) {
-      this.repository = repository;
+      this.#repository = repository;
     } else {
-      this.repository = this as any;
+      this.#repository = this as any;
     }
-    this.table = table;
-
-    // For some reason Object.assign(this, parent) didn't work in this case
-    // so doing all this manually:
-    this.$any = parent?.$any;
-    this.$all = parent?.$all;
-    this.$not = parent?.$not;
-    this.$none = parent?.$none;
-    this.$sort = parent?.$sort;
-
-    Object.assign(this, data);
```

Comment:
Hiding the internal ``filters`` into a private object instead of public ``class`` properties